core/services/pipeline/common.go

```diff
@@ -427,23 +427,34 @@ func UnmarshalTaskFromMap(taskType TaskType, taskMap interface{}, ID int, dotID
 	return task, nil
 }
 
-func CheckInputs(inputs []Result, minLen, maxLen, maxErrors int) ([]interface{}, error) {
-	if minLen >= 0 && len(inputs) < minLen {
-		return nil, errors.Wrapf(ErrWrongInputCardinality, "min: %v max: %v (got %v)", minLen, maxLen, len(inputs))
+// CheckInputsLen checks that inputs is error free and the expected length.
+func CheckInputsLen(inputs []Result, minLen, maxLen int) error {
+	if minLen > 0 && len(inputs) < minLen {
+		return errors.Wrapf(ErrWrongInputCardinality, "min: %v max: %v (got %v)", minLen, maxLen, len(inputs))
 	} else if maxLen >= 0 && len(inputs) > maxLen {
-		return nil, errors.Wrapf(ErrWrongInputCardinality, "min: %v max: %v (got %v)", minLen, maxLen, len(inputs))
+		return errors.Wrapf(ErrWrongInputCardinality, "min: %v max: %v (got %v)", minLen, maxLen, len(inputs))
 	}
-	var vals []interface{}
-	var errs int
+	return CheckInputs(inputs)
+}
+
+// CheckInputs returns ErrTooManyErrors if inputs contains an Error.
+func CheckInputs(inputs []Result) error {
 	for _, input := range inputs {
 		if input.Error != nil {
-			errs++
-			continue
+			return ErrTooManyErrors
```

Comment:
While you are already touching this logic, can you also propagate the error here?
Something like:
errors.Wrapf(ErrTooManyErrors, "Error in input index %v, detailed error = %v", i, input.Error)

Reply:
I started doing something like that but hesitated since I thought other logic might depend on detecting `ErrTooManyErrors`. I can try it out and see how the tests behave though. Another thought is that it might be redundant to include specific errors along with `ErrTooManyErrors` if they were already logged earlier.

Reply:
Yes I was suggesting ErrTooManyErrors error itself, with more details in the error message.
The tests should ideally be just checking for the error type as ErrTooManyErrors, not the detailed error text too.

Regarding logging, yes logging is definitely good, but ideally better to bubble up the right error to the caller.

Reply:
All checks should be using `errors.Is` or `err.Cause() == ` anyway so it should be safe to wrap otherwise checking code is broken.